src/eos_kolibri/eos_kolibri_migrate/main.py

```diff
@@ -16,119 +16,250 @@
-    if not source_path:
-        # TODO: discover Kolibri installs in desktop user homes
-        pass
+def kolibri_data_exists(kolibri_data_path):
+    data_file_paths = map(kolibri_data_path.joinpath, KOLIBRI_DATA_FILES)
+    return all(path.exists() for path in data_file_paths)
 
-    if not source_path:
-        logging.error("Must specify a valid source directory (set to {})".format(source_path))
-        return 1
 
-    if not target_path:
-        logging.error("Must specify a valid target directory")
-        return 1
+@dataclass(frozen=True)
+class MigrateCommand(object):
+    source_pwd: pwd.struct_passwd
+    is_you: bool
+    source_path: Path
+    target_path: Path
+    interactive: bool
 
-    if not target_path.exists():
-        target_path.mkdir(parents=True)
-    elif not os.listdir(target_path):
-        # Use the existing empty directory. Nothing to do here.
-        pass
-    elif replace:
-        backup_path = get_backup_path(target_path)
-        logging.info("Backing up original target directory to '{}'".format(backup_path))
-        target_path.rename(backup_path)
-        target_path.mkdir(parents=True)
-    else:
-        logging.error("Target directory ('{}') already exists".format(target_path))
-        return 1
+    def run(self):
+        # if not os.environ.get("KOLIBRI_USE_SYSTEM_INSTANCE"):
+        #     print("Your system is not configured to use system-wide Kolibri")
+        #     return 1
 
-    # Stop local Kolibri services and daemon, if running.
-    stop_kolibri_services()
-    terminate_kolibri_server_process()
+        if not kolibri_data_exists(self.source_path):
+            print(
+                "There is no Kolibri home in '{source_path}'".format(
+                    source_path=self.source_path.as_posix()
+                )
+            )
+            print("You are already using system-wide Kolibri")
+            return 0
+
+        print(
+            "Detected personal Kolibri home in '{source_path}'".format(
+                source_path=self.source_path.as_posix()
+            )
+        )
 
-    logging.info("Copying files from '{}'".format(source_path))
+        if not os.access(self.source_path.parent, os.W_OK):
+            print(
+                "The personal Kolibri home '{source_path}' cannot be renamed".format(
+                    source_path=self.source_path.as_posix()
+                )
+            )
+            print("Please run this program like 'sudo eos-kolibri-migrate'")
+            return 1
+
+        if not os.access(self.target_path.parent, os.W_OK):
+            print(
+                "The system Kolibri home '{target_path}' cannot be renamed".format(
+                    target_path=self.target_path.as_posix()
+                )
+            )
+            print("You may need to run this program like 'sudo eos-kolibri-migrate'")
+            return self.__run_enable_without_copy(True)
```

Comment:
Looking at `__run_enable_without_copy`, seems to me that it just backup the source folder, why does do that? subsequent runs of this command won't find the user Kolibri home. Shouldn't we just fail in this case?

Reply:
I tidied this up a bunch with the switch to click. Please do let me know if it reads more logically now :)